### PR TITLE
[close #619] Shutdown recycler when closing ChannelFactory to avoid resource leak

### LINKED ISSUE #619: ScheduledExecutorService may not be shutdown in ChannelFactory
### 1. Describe the bug
```
public class ChannelFactory implements AutoCloseable {
    private final ScheduledExecutorService recycler = Executors.newSingleThreadScheduledExecutor();
     ....
     public void close() {
     ....
      if (certContext != null) {
        recycler.shutdown();                                      // recycler would not shutdown when TLS is disable
        if (certWatcher != null) {
          certWatcher.close();
        }
      }
    }
}
```

### PR DESCRIPTION
**Pull request by Daemonxiao**
Signed-off-by: Daemonxiao <735462752@qq.com>

### What problem does this PR solve?

Issue Number: close #619

Because the `recycler` in `ChannelFactory` does not set any task when TLS is turned off, the process will get stuck on exit.

### What is changed and how does it work?
When we don't use TLS, `recycler` should be `null`.

### Code changes

- Has exported function/method change

### Side effects

- NO side effects

### Related changes

- Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by iosmanthus** [comment]
@Daemonxiao Could you create an issue with this pull request? The PR title checker will check if the PR is related to an issue.

File: src/main/java/org/tikv/common/util/ChannelFactory.java [hunk 1]
```diff
@@ -361,7 +364,7 @@ public void close() {
     connPool.clear();
 
     if (certContext != null) {
-      recycler.shutdown();
+      recycler.shutdownNow();
```
Comment (iosmanthus) [inline src/main/java/org/tikv/common/util/ChannelFactory.java:367]: What about removing the `if (certContext != null)`:

```java
if (recycler != null ) {
    // shutdown recycler
}

if (certWatcher != null) {
    // shutdown certWatcher
}
```

Comment (Daemonxiao) [reply src/main/java/org/tikv/common/util/ChannelFactory.java:367]: @iosmanthus Thanks for your review. OK, I will change it.

**Comment by iosmanthus** [comment]
The tests for `ChannelFactory` might leak executors because some `certWatcher` or `ChannelFactory` are not closed automatically with the try-with-resource block. Could you fix it within this pull request? I think these two issues are related. Rest LGTM

**Review by zhangyangyu — APPROVED** [review]

**Review by iosmanthus — APPROVED** [review]
LGTM

**Pull request merged**